src/vector/faer_serial.rs

```diff
@@ -0,0 +1,191 @@
+use std::ops::{Div, Mul, MulAssign};
+
+use faer::{unzipped, zipped, Col, ColMut, ColRef};
+
+use crate::{scalar::Scalar, scalar::Scale, IndexType};
+
+use super::{Vector, VectorCommon, VectorIndex, VectorView, VectorViewMut};
+impl<'a, T: Scalar> Mul<Scale<T>> for ColRef<'a, f64> {
+    type Output = Col<f64>;
+    fn mul(self, rhs: Scale<T>) -> Self::Output {
```

Comment:
these three mul's can be combined into a single macro

Reply:
and the div below

Reply:
I think only the two first Mul's can be combined here. The third one is not expecting a lifetime. and for the Div one the function implementation is too different, I should be able to pass an anonymous function to the macro but that seems too convoluted, no?

Reply:
ok, I don't want to make it too convoluted, so feel free to leave as is